core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -127,49 +130,50 @@ private void stopContainer(String containerId, String imageName) {
     /**
      * Register a network to be cleaned up at JVM shutdown.
      *
-     * @param networkName   the image name of the network
+     * @param id   the ID of the network
      */
-    public void registerNetworkForCleanup(String networkName) {
-        registeredNetworks.add(networkName);
+    public void registerNetworkIdForCleanup(String id) {
+        registeredNetworks.add(id);
     }
 
     /**
      * Removes any networks that contain the identifier.
-     * @param identifier
+     * @param id
      */
-    public void removeNetworks(String identifier) {
-      removeNetwork(identifier);
+    public void removeNetworkById(String id) {
+      removeNetwork(id);
     }
 
-    private void removeNetwork(String networkName) {
+    private void removeNetwork(String id) {
         try {
-            try {
```

Comment:
I think this block should do the job when parameter is id, and "list with filter" is not required anymore